src/components/TokenClaimDialog.tsx

```diff
@@ -49,6 +54,9 @@ export default function TokenClaimDialog({
   const { t } = useTranslation()
   const { chainId } = useActiveWeb3React()
   const basicPools = useContext(BasicPoolsContext)
+  const userState = useContext(UserStateContext)
+  const gaugeData = useContext(GaugeContext)
+  const gauges = Object.values(gaugeData.gauges)
```

Comment:
this will be typed as `(Gauge | undefined)[]`. Since we know that anything returned by `Object.values` will be a `Gauge`, you can cast this `as Gauge[]` which should make things a little easier later on. 

The reason it's `Gauge | undefined` is because `gaugeData.gauges[someString]` may be undefined.

Reply:
Also will this trigger rerenders? `gaugeData` will be the same (`===`) object between renders, but I'm guessing `Object.values` creates a new object each time.

Reply:
I don't see it triggering re-renders.

Reply:
>The reason it's Gauge | undefined is because gaugeData.gauges[someString] may be undefined.

I was wondering about this.